TumbleBitSetup/RsaKey.cs

```diff
@@ -68,6 +79,18 @@ internal byte[] Decrypt(byte[] encrypted)
             return engine.ProcessBlock(encrypted, 0, encrypted.Length);
         }
 
+        public PoupardSternProof CreatePoupardSternProof(PoupardSternSetup setup)
```

Comment:
If you want to include to BC, then you will need a separate PR for BC which respect their design guideline. When I use BC with TB, I just hid everything with my own custom classes (RSAKey 
 and RSAPubKey class do not belong to BC), because I am working on a different abstraction layer, for developers.  
BC has a different abstraction layer, oriented for cryptographers.

The RSAKey and RSAPubKey are a developer oriented abstraction layer, which BC do not have.

In other words, RSAPubKey and RSAKey would not be included in BC as it is for developer, not cryptographer.

Now the question is: do you want to make the design of this repo adhere to BC guidelines to facilitate BC integration later? Or do you want to make it usable for developers?

Note that I tried to not add dependencies from PermutationTest and PoupartSternTest to RSAKey and RSAPubKey (and actually your classes already depends on RSAKey RSAPubKey, this is fishy design as the cryptographer layer should not depends on the dev layer, we should fix that in later PR.

On my side I this we should keep the PermutationTest and PoupartSternTest classes copy/pastable to BC, while still having RSAKey and RSAPubKey classes to show how to use it correctly to devs.